### ISSUE TITLE
Can not send test mail if wp-config.php contains advanced plugin confiuration

### ISSUE THREAD

**Issue by Peneheals** [awesomemotive/WP-Mail-SMTP#47]
## Plugin version 

1.4.1

## Current Behavior

After a clean WP install, if I download and enable the plugin, and set it up through the plugin's dashboard, it works fine, I can send the test mail. As it is adviced, I set two variable in the wp-config.php (WPMS_ON to true and the WPMS_SMTP_PASS).

But if I make again a fresh WP install, and do the same (DLing and installing the plugin), but I configure almost everything in the wp-config.php except the WPMS_MAIL_FROM_NAME and the WPMS_SMTP_USER, and after this I configure the WPMS_MAIL_FROM_NAME and the WPMS_SMTP_USER on the dashboard I am not able to send the test e-mail, there is a text next to the send button:

> You cannot send an email. Mailer is not properly configured. Please check your settings.

## Expected Behavior

I should be able to send (test) messages when I configure the plugin through wp-config.php and the plugin's dashboard (or any kind of mix of them, if my settings are correct).

## Possible Solution

Maybe I'm wrong, maybe not:

It looks like, if I'm starting to comment out one by one the options from my wp-config.php, the problem is solved if I comment out WPMS_SMTP_PORT and set it through the plugin's admin dashboard. (Maybe I'm mistaken and I have to configure the WPMS_SMTP_HOST too.)

## Steps to Reproduce (for bugs)

1. Install a clean WP&WP Mail SMTP and enable the plugin (do not configure them on the dashboard).
2. Set everything in the wp-config.php except WPMS_MAIL_FROM_NAME and WPMS_SMTP_USER, so those:

- WPMS_ON
- WPMS_MAIL_FROM
- WPMS_MAIL_FROM_FORCE
- WPMS_MAIL_FROM_NAME_FORCE
- WPMS_MAILER
- WPMS_SET_RETURN_PATH
- WPMS_SMTP_HOST
- WPMS_SMTP_PORT
- WPMS_SSL
- WPMS_SMTP_AUTOTLS
- WPMS_SMTP_AUTH
- WPMS_SMTP_PASS

<img width="524" alt="all_set" src="https://user-images.githubusercontent.com/7722456/54715987-01710c00-4b55-11e9-984c-0096e745e340.png">

3. Set WPMS_MAIL_FROM_NAME and WPMS_SMTP_USER on the admin dashboard. Save it.

![wp_mail_smtp_dashboard_after_save](https://user-images.githubusercontent.com/7722456/54716141-6298df80-4b55-11e9-940f-0ae5c249d8e7.png)

4. Try to send a test e-mail. You can not.

<img width="710" alt="cantsend" src="https://user-images.githubusercontent.com/7722456/54716173-7a706380-4b55-11e9-8525-920f9ceb7ddd.png">

## WP-CLI outputs

Outputs of wp-cli's `wp option pluck wp_mail_smtp smtp --format=json` command:

```
# Plugin not installed:
(empty)
# Plugin installed and enabled:
{"autotls":true}
# Everything set in wp-config except smtp username (that through dashboard):
{"autotls":true,"user":"XYZ","auth":true}
# And so on. If I comment out the port from wp-config and set on the dashboard, I can send test e-mail:
{"autotls":true,"user":"XYZ","auth":true,"host":"email-smtp.eu-west-1.amazonaws.com","port":587,"encryption":"tls"}
```

## Related issues: 

https://github.com/awesomemotive/WP-Mail-SMTP/issues/18
https://github.com/awesomemotive/WP-Mail-SMTP/issues/27
https://github.com/awesomemotive/WP-Mail-SMTP/issues/18#issuecomment-432735219

## Final words

Of course with the same settings I'm able to send test letters if I configure everything on the dashboard, and nothing in the wp-config.php, and I'm able too if I configure the WPMS_ON and WPMS_SMTP_PASS in the wp-config.php and everything else on the dashboard.

**Comment by larsha** [comment]
Having same issue here!

**Comment by slaFFik** [comment]
Seems to be fixed in #35.

**Comment by pkarjala** [comment]
FYI, I am still seeing this behavior in 1.4.2.
